src/App.tsx

```diff
@@ -1,26 +1,279 @@
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+            data-cy="ToggleAllButton"
+          />
+
+          <form onSubmit={handleSubmit}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={title}
+              onChange={event => setTitle(event.target.value)}
+              ref={inputRef}
+              disabled={tempTodo !== null}
+            />
+          </form>
+        </header>
+
+        {!!todos.length && (
+          <>
+            <section className="todoapp__main" data-cy="TodoList">
+              {visibleTodos.map(todo => (
+                <TodoItem
+                  key={todo.id}
+                  todo={todo}
+                  onDelete={() => handleDeletingTodo(todo.id)}
+                  loading={processingIds.includes(todo.id)}
+                  editInputRef={editInputRef}
+                  setEditId={setEditId}
+                  editId={editId}
+                  setEditTitle={setEditTitle}
+                  editTitle={editTitle}
+                  handleSubmitEdit={handleSubmitEdit}
+                  handleCancelEdit={handleCancelEdit}
+                />
+              ))}
+
+              {tempTodo && <TodoItem todo={tempTodo} loading={true} />}
+            </section>
+
+            <footer className="todoapp__footer" data-cy="Footer">
+              <span className="todo-count" data-cy="TodosCounter">
+                {`${activeTodos.length} items left`}
+              </span>
+
+              <nav className="filter" data-cy="Filter">
+                <a
+                  href="#/"
+                  data-cy="FilterLinkAll"
+                  onClick={() => setStatus(Filter.all)}
+                  className={classNames('filter__link', {
+                    selected: status === Filter.all,
+                  })}
+                >
+                  All
+                </a>
+
+                <a
+                  href="#/active"
+                  data-cy="FilterLinkActive"
+                  onClick={() => setStatus(Filter.active)}
+                  className={classNames('filter__link', {
+                    selected: status === Filter.active,
+                  })}
+                >
+                  Active
+                </a>
+
+                <a
+                  href="#/completed"
+                  data-cy="FilterLinkCompleted"
+                  onClick={() => setStatus(Filter.completed)}
+                  className={classNames('filter__link', {
+                    selected: status === Filter.completed,
+                  })}
+                >
+                  Completed
+                </a>
+              </nav>
+
+              {!!completedTodos.length && (
```

Comment:
the filter should always be centered, even if `clear completed` button is not visible